foresight/environment/azure/azure_environment_info_provider.py

```diff
@@ -0,0 +1,62 @@
+from foresight.environment.git.git_helper import GitHelper
+from foresight.environment.environment_info import EnvironmentInfo
+from foresight.utils.test_runner_utils import TestRunnerUtils
+import os, logging
+from foresight.utils.generic_utils import print_debug_message_to_console
+
+LOGGER = logging.getLogger(__name__)
+
+class AzureEnvironmentInfoProvider:
+
+    ENVIRONMENT = "Azure"
+    BUILD_BUILDID_ENV_VAR_NAME = "BUILD_BUILDID"
+    BUILD_REPOSITORY_NAME_ENV_VAR_NAME = "BUILD_REPOSITORY_NAME"
+    BUILD_REPOSITORY_URI_ENV_VAR_NAME = "BUILD_REPOSITORY_URI"
+    BUILD_SOURCEBRANCHNAME_ENV_VAR_NAME = "BUILD_SOURCEBRANCHNAME"
+    BUILD_SOURCEVERSION_ENV_VAR_NAME = "BUILD_SOURCEVERSION"
+    BUILD_SOURCEVERSIONMESSAGE_ENV_VAR_NAME = "BUILD_SOURCEVERSIONMESSAGE"
+    INVOCATION_ID_ENV_VAR_NAME = "INVOCATION_ID"
+    BUILD_REPOSITORY_PROVIDER_ENV_VAR_NAME = "BUILD_REPOSITORY_PROVIDER"
+
+    @classmethod
+    def get_test_run_id(cls, repo_url, commit_hash, environment):
+        configured_test_run_id = TestRunnerUtils.get_configured_test_run_id()
+        if configured_test_run_id:
+            return configured_test_run_id
+        build_number = os.getenv(cls.BUILD_BUILDID_ENV_VAR_NAME)
+        invocation_id = os.getenv(cls.INVOCATION_ID_ENV_VAR_NAME)
+        if build_number or invocation_id:
+            return TestRunnerUtils.get_test_run_id(environment, repo_url, commit_hash, invocation_id + "-" + build_number)
```

Comment:
if build_number or invocation_id is null, **invocation_id + "-" + build_number** expression throws exception.  Null check should be done.

Reply:
fixed.